test/nitro/claim-ts.test.ts

```diff
@@ -12,189 +12,223 @@ describe("claim (typescript)", function () {
-  const TARGET_CHANNEL_ADDRESS = "0x080678731247781ff0d57c649b6d0ad1a0620df0"; // At some point in the full claim operation, the outcome of this channel must be read and checked
-
-  it("Can claim with no exit requests", async function () {
-    const initialOutcome: Exit = [
+  const CHANNEL_1 = "0x080678731247781ff0d57c649b6d0ad1a0620df0"; // At some point in the full claim operation, the outcome of this channel must be read and checked
+  const CHANNEL_2 = "0x27592B3827907B54684E4f9da3d988263828893D"; // At some point in the full claim operation, the outcome of this channel must be read and checked
+  const I_ADDRESS = "0x7ab853663C531EaA080d84091AD0E0e985c688C7";
+
+  const createGuarantee = (
+    guarantees: ["C1" | "C2", BigNumberish, Array<"A" | "B" | "I">][]
+  ): Exit => {
+    return [
       {
         asset: ZERO_ADDRESS,
         data: "0x",
-        allocations: [
-          {
-            destination: A_ADDRESS,
-            amount: "0x05",
-            callTo: ZERO_ADDRESS,
-            data: "0x",
-          },
-          {
-            destination: B_ADDRESS,
-            amount: "0x05",
-            callTo: ZERO_ADDRESS,
-            data: "0x",
-          },
-        ],
+        allocations: guarantees.map((g) => {
+          const guaranteeList = g[2].map(
+            (p) =>
+              (p === "A"
+                ? A_ADDRESS
+                : p === "B"
+                ? B_ADDRESS
+                : I_ADDRESS) as string
+          );
+
+          return {
+            destination: g[0] === "C1" ? CHANNEL_1 : CHANNEL_2,
+            amount: BigNumber.from(g[1]).toHexString(),
+            callTo: MAGIC_VALUE_DENOTING_A_GUARANTEE,
+            data: encodeGuaranteeData(...guaranteeList),
+          };
+        }),
       },
     ];
-
-    const guarantee: Exit = [
+  };
+  const createOutcome = (
+    allocations: ["A" | "B" | "I", BigNumberish][]
+  ): Exit => {
+    return [
       {
         asset: ZERO_ADDRESS,
         data: "0x",
-        allocations: [
-          {
-            destination: TARGET_CHANNEL_ADDRESS,
-            amount: "0x00",
-            callTo: MAGIC_VALUE_DENOTING_A_GUARANTEE,
-            data: encodeGuaranteeData(B_ADDRESS, A_ADDRESS),
-          },
-        ],
+        allocations: allocations.map((a) => ({
+          destination:
+            a[0] === "A" ? A_ADDRESS : a[0] === "B" ? B_ADDRESS : I_ADDRESS,
+          amount: BigNumber.from(a[1]).toHexString(),
+          callTo: ZERO_ADDRESS,
+          data: "0x",
+        })),
       },
     ];
+  };
+
+  it("Can handle an underfunded claim", async function () {
+    const initialOutcomeForTargetChannel = createOutcome([
+      ["A", "0x05"],
+      ["B", "0x05"],
+      ["I", "0x0A"],
+    ]);
+    const initialOutcomeForAnotherChannel = createOutcome([["A", "0x03"]]);
+    const guarantee = createGuarantee([
+      ["C2", "0x03", ["A"]],
+      ["C1", "0x0A", ["A", "I", "B"]],
+    ]);
```

Comment:
I like this notation a lot, it's easy to parse the important information.

Any reason why you chose C2 as the higher priority and C1 as the lower priority?

Reply:
No strong reason. 

In my mind, `C1` is the "main" channel (with allocations to A,I,B) and we're using C2 ahead of it to test the underfunded claim scenario.  It keeps it consistent-ish with other tests that use `C1` as the only channel.